web_calendar_custom/models/res_config_settings.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    calendar_start_time = fields.Float('Calendar Start Time')
+    calendar_end_time = fields.Float('Calendar End Time')
+    calendar_start_work_time = fields.Float('Calendar Start Working Time')
+    calendar_first_day_of_week = fields.Selection([('0', 'Sunday'),
+                                                   ('1', 'Monday'),
+                                                   ('2', 'Tuesday')],
+                                                  default='1')
+    calendar_row_duration = fields.Selection([('0.25', '00:15'),
+                                              ('0.5', '00:30'),
+                                              ('1.0', '01:00')],
+                                             default='0.25')
+    is_weekend_active = fields.Boolean('Calendar Weekends', default=True)
+    is_event_overlap = fields.Boolean('Calendar Overlap Events', default=False)
+
+    _sql_constraints = [
+        ('check_calendar_start_and_end_time',
+         "CHECK ((calendar_start_time < calendar_end_time) AND"
+         "(calendar_start_time >= 0) AND ( calendar_start_time < 24) AND"
+         "(calendar_end_time >= 0) AND ( calendar_end_time < 24)) ",
+         _('Start time must be before end time')),
+        ('check_calendar_start_work_time',
+         "CHECK ((calendar_start_time < calendar_start_work_time) AND"
+         "(calendar_start_work_time < calendar_end_time) AND"
+         "( calendar_start_time < 24) AND (calendar_end_time >= 0))",
+         _('Start work time must be set between start and end time')),
+    ]
+
+    @api.model
+    def get_values(self):
+        res = super(ResConfigSettings, self).get_values()
+        res.update(
+            calendar_start_time=self.env[
```

Comment:
We can try avoid repeating ourselves by `model_ir_config_parameter = self.env['ir.config_parameter'].sudo()` and then use that for all our calls.